tests/integration/internal/stepfuncs/traces.go

```diff
@@ -0,0 +1,54 @@
+package stepfuncs
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/tracesgenerator"
+)
+
+// Generate logsCount logs using the designated implementation type:
+// either deployment or a daemonset.
```

Comment:
The comment says we can use either a Deployment or Daemonset, but the implementation just uses a Daemonset unconditionally. Is there any benefit to using a Daemonset here to begin with?

Reply:
To be honest, it's a leftover from copying and paste :)
I was hoping that someone would stratify this in a review, but I forgot to mention that in a comment, sorry.
Is there some overhead related to using Daemonset? I guess that changing to deployment won't hurt.

Reply:
There isn't overhead, but DaemonSets and Deployments have different semantics. In particular, a DaemonSet starts a Pod for each Node in the cluster, while a Deployment will start as many as you tell it to. The kind cluster used for tests only has one Node, so they're effectively equivalent in this case, but the test would start failing if someone adds more Nodes, which is not ideal.

In general, I don't think there's any reason to use a DaemonSet here. @mat-rumian WDYT?

Reply:
As we use the same version of OT Distro for all our deployments I think it's not necessary to run tests with Daemonset. Single deployment testing traces pipeline should be enough :)

Reply:
I fixed this. However, I believe that there is one difference: Deployment causes the generator to run multiple times, while DaemonSet runs only once and terminate. Because of that, these tests are not testing a case where some of the traces are being dropped, because the data is no longer limited.

Reply:
If you don't want the Deployment to restart the container, you can use the same trick we used for log generation here: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/887e2b312e99b34c0e7f7e76a96ae2e5eac7295c/tests/integration/internal/logsgenerator/logsgenerator.go#L85, which is basically just adding a sleep command.